### PR TITLE
dijkstra algo added

### PR DESCRIPTION
**Pull request by Yashgabani845**
## Related Issue
Fixes : #562 

## Description

Added explanation about dijkstra algorithm 

1. with 4 programming language code,
2. approch
3. problems
4. complexity
5. pseudo code
6. conclusion sections 

also created new section of algorithms so added category file too
## Type of PR

- [x] Feature enhancement
- [x] Documentation update

## Screenshots / Videos (if applicable)
Screenshot (468)
Screenshot (469)
Screenshot (470)
Screenshot (471)
[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

sir i had one doubt
when i use any code viewer of python,js,cpp then it shows code with few colours and format 
but in java code viewer it is showing plan text no colours
i have searched for this but not getting 
can you please tell me the reason or solution if you know
## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: dsa/Algorithms/dijkstra.md [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+## Advanced Topics and Optimizations
+
+### Bidirectional Dijkstra
+
+Bidirectional Dijkstra runs two simultaneous searches: one forward from the source and one backward from the target. This can significantly speed up the search in large graphs.
+
+### Time Complexity
+
+The time complexity of Dijkstra's Algorithm depends on the data structures used:
+- Using a simple list: O(V^2)
```
Comment (Ajay-Dhangar) [inline dsa/Algorithms/dijkstra.md:349]: replace `O(V^2)` to `$O(V^2)$`

File: dsa/Algorithms/dijkstra.md [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+- Using a binary heap (priority queue): O((V + E) log V)
```
Comment (Ajay-Dhangar) [inline dsa/Algorithms/dijkstra.md:350]: replace `O((V + E) log V)` to `$O((V + E) log V)$`

File: dsa/Algorithms/dijkstra.md [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+- Using a Fibonacci heap: O(V log V + E)
```
Comment (Ajay-Dhangar) [inline dsa/Algorithms/dijkstra.md:351]: replace `O(V log V + E)` to `$O(V log V + E)$`

File: dsa/Algorithms/dijkstra.md [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+
+### Path Reconstruction
+
+To reconstruct the shortest path from the source to a target node, we can backtrack from the target node using the `previous_nodes` dictionary.
+
+## Conclusion
+
+In this tutorial, we covered the fundamentals of Dijkstra's Algorithm, its implementation in Python, Java, C++, and JavaScript, and various optimizations and applications. Dijkstra's Algorithm is a powerful tool for finding the shortest path in graphs and is widely used in numerous domains. By mastering this algorithm, you can effectively solve a variety of shortest path problems in your projects.
+
+Happy coding!
```
Comment (Ajay-Dhangar) [inline dsa/Algorithms/dijkstra.md:365]: remove line `Happy coding!`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
now update content based on suggestions

**Comment by Yashgabani845** [comment]
@Ajay-Dhangar 
sorry for repeating same mistake
will take care about it

**Review by Ajay-Dhangar — APPROVED** [review]
done

**Pull request merged**